opm/autodiff/SimulatorFullyImplicitBlackoilOutput.hpp

```diff
@@ -375,5 +378,179 @@ namespace Opm
     }
 
 
+
+
+
+    namespace detail {
+
+        struct WriterCall : public ThreadHandle :: ObjectInterface
+        {
+            BlackoilOutputWriter& writer_;
+            std::unique_ptr< SimulatorTimerInterface > timer_;
+            const SimulationDataContainer state_;
+            const WellState wellState_;
+            std::vector<data::CellData> simProps_;
+            const bool substep_;
+
+            explicit WriterCall( BlackoilOutputWriter& writer,
+                                 const SimulatorTimerInterface& timer,
+                                 const SimulationDataContainer& state,
+                                 const WellState& wellState,
+                                 const std::vector<data::CellData>& simProps,
+                                 bool substep )
+                : writer_( writer ),
+                  timer_( timer.clone() ),
+                  state_( state ),
+                  wellState_( wellState ),
+                  simProps_( simProps ),
+                  substep_( substep )
+            {
+            }
+
+            // callback to writer's serial writeTimeStep method
+            void run ()
+            {
+                // write data
+                writer_.writeTimeStepSerial( *timer_, state_, wellState_, simProps_, substep_ );
+            }
+        };
+
+
+
+
+        template<class Model>
+        std::vector<data::CellData> getCellData(const Model& model,
+                const RestartConfig& restartConfig,
+                const int reportStepNum) {
+            std::vector<data::CellData> simProps;
+
+
+            std::map<const char*, int> outKeywords {
+                {"ALLPROPS", 0},
+
+                {"BG", 0},
+                {"BO", 0},
+                {"BW", 0},
+
+                {"CONV", 0},
+                {"DEN", 0},
+
+                {"KRG", 0},
+                {"KRO", 0},
+                {"KRW", 0},
+
+                {"RVSAT", 0},
+                {"RSSAT", 0},
+
+                {"NORST", 0},
+                {"PBPD", 0},
+                {"VISC", 0}
+            };
+
+            //Get the value of each of the keys
+            for (auto& keyValue : outKeywords) {
+                keyValue.second = restartConfig.getKeyword(keyValue.first, reportStepNum);
+            }
+
+            //Postprocess some of the special keys
+            if (outKeywords["ALLPROPS"] > 0) {
```

Comment:
I received ALLPROPS, but not the others. I'll double check, maybe I did something on my side...